pkg/internal/transform/spanner.go

```diff
@@ -107,16 +109,19 @@ func extractIP(b []uint8, size int) string {
 	return net.IP(b[:size]).String()
 }
 
-func extractTraceID(traceparent [55]byte) string {
+func extractTraceparent(traceparent [55]byte) (string, string, string) {
```

Comment:
Do you think it's better if we moved this logic now completely to the traces code? We do this extra allocation here of technically almost the same size as the incoming array, and it's all wasted if we are only producing metrics. We can simply let the original byte array get all the way to the traces code and have the array slicing done there.